tests/integration_tests/testsuite/ovc_tests/a_basic/vms_tests.py

```diff
@@ -109,3 +109,122 @@ def test002_create_vms_with_correct_params(self):
+        self.vdcusers[cls.vdcuser] = {'openvcloud': self.openvcloud,
+                                      'provider': 'itsyouonline',
+                                      'email': '%s@test.com' % self.random_string(),
+                                      'groups': ['user']}
+        cls.vdcusers = self.vdcusers
+        cls._testID = self._testID
+        cls.temp_actions = {'account': {'actions': ['install']},
+                            'vdcuser': {'actions': ['install']},
+                            'vdc': {'actions': ['install']},
+                            'node': {'actions': ['install']}}
+        self.log('Create vm, should succeed')
+        res = self.create_vm(accounts=cls.accounts, cloudspaces=cls.cloudspaces,
+                             vms=cls.vms, temp_actions=cls.temp_actions)
+        self.wait_for_service_action_status(self.vm1, res[self.vm1]['install'])
+        self.CLEANUP["accounts"].append(cls.acc1)        
+
+    def tearDown(self):
+        pass
+
+    def test001_adding_and_deleting_portforward(self):
+        """ ZRT-OVC-012
+        *Test case for adding and deleting portforward.*
+
+        **Test Scenario:**
+
+        #. Create a vm[vm1], should succeed.
+        #. Create a portforward for [vm1], should succeed.
+        #. Check that portforward created, should succeed.
+        #. Delete the portforward created, should succeed.
+        #. Check that portforward deleted, should succeed. 
+        """
+        self.log('%s STARTED' % self._testID)
+
+        self.log("Create portforward for [vm1], should succeed. ")        
+        public_port = randint(1000, 60000)
+        local_port = 22        
+        temp_actions = {'node': {'actions': ['portforward_create'], 'service': self.vm1, 
+                                 'args': {'ports': {'source': public_port, 'destination': local_port}}}}
+        res = self.create_vm(accounts=self.accounts, cloudspaces=self.cloudspaces,
+                             vms=self.vms, temp_actions=temp_actions)
+        self.wait_for_service_action_status(self.vm1, res[self.vm1]['portforward_create'])
+
+        self.log(" Check that portforward created, should succeed.")
+        self.assertTrue(self.get_portforward_list(self.cs1, self.vm1))
+
+        self.log("Delete the portforward created, should succeed")
+        temp_actions = {'vdc': {'actions': ['portforward_delete'], 'service': self.vm1, 
+                        'args': {'ports': {'source': public_port, 'destination': local_port}}}}
+        res = self.create_vm(accounts=self.accounts, cloudspaces=self.cloudspaces,
+                             vms=self.vms, temp_actions=temp_actions)
+        self.wait_for_service_action_status(self.vm1, res[self.vm1]['portforward_delete'])
+
+        self.log('Check that the portforward  deleted from cloudspace [CS1] successfully , should succeed.')
+        self.assertFalse(self.get_portforward_list(self.cs1, self.vm1))
```

Comment:
in which case the get_portforward_list will return false .. please check on the specific port forward anyways